hbase-server/src/main/java/org/apache/hadoop/hbase/security/access/SnapshotScannerHDFSAclController.java

```diff
@@ -610,7 +609,8 @@ private Set<String> filterUsersToRemoveNsAccessAcl(Table aclTable, TableName tab
           PermissionStorage.isGlobalEntry(entry)
             || (PermissionStorage.isNamespaceEntry(entry)
               && Bytes.equals(PermissionStorage.fromNamespaceEntry(entry), namespace))
-            || (!Bytes.equals(tableName.getName(), entry)
+            || (PermissionStorage.isTableEntry(entry)
```

Comment:
So this is the problem we want to fix? Mind explaining more?

Reply:
Yes, this is the problem.
The `userEntries` from hbase:acl contains namespaces (beginning with '@' like '@ns1') and tables, before we call `TableName.valueOf(entry)`, must make sure it is a table name and not a namespace name. If a namespace name goes there, `TableName.valueOf(entry)` will throw IllegalArgumentException since char '@'.